tools/cxbe/Exe.cpp

```diff
@@ -45,11 +45,11 @@ Exe::Exe(const char *x_szFilename)
 
     FILE *ExeFile = fopen(x_szFilename, "rb");
 
-    // verify Exe file was opened
-    if(ExeFile == 0)
+    // verify Exe file was opened successfully
+    if(ExeFile == NULL)
     {
-        SetError("Could not open Exe file.", true);
-        return;
+        SetError("Could not open Exe file", true);
+        goto cleanup;
```

Comment:
nit: Using `goto`'s.

Reply:
I assume this is aimed at the whole "do not use `goto`" debate? Most of that `goto`-criticsm is very old and was meant to advocate the use of structured code (`while` and `for`) which were being introduced around that time.
`goto` is still useful for situations like this. - the readability would only get worse if code was encapsulated in tiny helper functions, or even in a `while` loop using `break` statements.

C++ also introduced exceptions to avoid `goto`, but this codebase does not use exceptions. In fact, a lot of organizations do not use exceptions for different reasons.

This codebase already uses `goto` in many places.
The error that is being fixed here was caused by erroneous use of `return` - all other instances in this function use `goto` (which close the open file descriptor, and print the error message).

If you think we should avoid `goto` in this repository, I'd suggest to open an issue about it.
Personally, I think it's fine to use it in cases like this.

---

However, there is actually a bug with this, because the `cleanup` does unconditionally close the file descriptor. I'll fix this by adding a check so it's not closed if it's already `NULL`.